packages/x-date-pickers/src/internals/hooks/useField/useFieldInternalPropsWithDefaults.ts

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import type { FieldChangeHandler, UseFieldInternalProps } from './useField.types';
+import {
+  PickerAnyManager,
+  PickerManagerError,
+  PickerManagerFieldInternalProps,
+  PickerManagerFieldInternalPropsWithDefaults,
+  PickerManagerValue,
+} from '../../models';
+import { PickerContext } from '../../../hooks/usePickerContext';
+import { useLocalizationContext } from '../useUtils';
+
+export const PickerFieldPrivateContext = React.createContext<PickerFieldPrivateContextValue | null>(
+  null,
+);
+
+/**
+ * Applies the default values to the field internal props.
+ * This is a temporary hook that will be removed during a follow up when `useField` will receive the internal props without the defaults.
+ * It is only here to allow the migration to be done in smaller steps.
+ */
+export function useFieldInternalPropsWithDefaults<TManager extends PickerAnyManager>({
+  manager,
+  internalProps,
+}: {
+  manager: TManager;
+  internalProps: PickerManagerFieldInternalProps<TManager>;
+}): PickerManagerFieldInternalPropsWithDefaults<TManager> {
+  const localizationContext = useLocalizationContext();
+  const privateContextValue = React.useContext(PickerFieldPrivateContext);
+  // TODO: Replace with useNullablePickerContext
+  const publicContextValue = React.useContext(PickerContext);
+
+  const internalPropsWithDefaultsFromContext = React.useMemo(() => {
+    // If one of the context is null, the other always will be null as well.
+    if (privateContextValue == null || publicContextValue == null) {
+      return internalProps;
+    }
+    const handleChange: FieldChangeHandler<
```

Comment:
I can probably do something cleaner once we have a true refactoring of `useField`, but for now I avoided touching to many pieces and I just made sure we always return a function that has the same signature as `props.onChange`